tests/neo.UnitTests/Consensus/UT_Consensus.cs

```diff
@@ -100,35 +98,234 @@ public void ConsensusService_Primary_Sends_PrepareRequest_After_OnStart()
+            var rmPayload = (ConsensusPayload)backupOnRecoveryMessageAfterCommit.Inventory;
+            RecoveryMessage rmm = (RecoveryMessage)rmPayload.ConsensusMessage;
+            // =============================================
+
+            mockContext.Object.CommitPayloads[0] = null;
+            Console.WriteLine("\nCN5 simulation time");
+            actorConsensus.Tell(GetCommitPayloadModifiedAndSignedCopy(commitPayload, 4, kp_array[4], updatedBlockHashData));
+
+
+            Console.WriteLine($"\nFocing block PrevHash to UInt256.Zero {mockContext.Object.Block.GetHashData().ToScriptHash()}");
+            mockContext.Object.Block.PrevHash = UInt256.Zero;
+            // Payload should also be forced, otherwise OnConsensus will not pass
+            commitPayload.PrevHash = UInt256.Zero;
+            Console.WriteLine($"\nNew Hash is {mockContext.Object.Block.GetHashData().ToScriptHash()}");
+
+            Console.WriteLine($"\nForcing block VerificationScript to {updatedContract.Script.ToScriptHash()}");
+            mockContext.Object.Block.Witness = new Witness { };
+            mockContext.Object.Block.Witness.VerificationScript = updatedContract.Script;
+            Console.WriteLine($"\nUpdating BlockBase Witness scripthash  is {mockContext.Object.Block.Witness.ScriptHash}");
+            Console.WriteLine($"\nNew Hash is {mockContext.Object.Block.GetHashData().ToScriptHash()}");
+
+            Console.WriteLine("\nCN6 simulation time");
+            // Here we used modified mockContext.Object.Block.GetHashData().ToScriptHash() for blockhash
+            actorConsensus.Tell(GetCommitPayloadModifiedAndSignedCopy(commitPayload, 5, kp_array[5], mockContext.Object.Block.GetHashData()));
+
+            Console.WriteLine("\nWait for subscriber Local.Node Relay");
+            var onBlockRelay = subscriber.ExpectMsg<LocalNode.Relay>();
+            Console.WriteLine("\nAsserting time was Block...");
+            var utBlock = (Block)onBlockRelay.Inventory;
+
+            Console.WriteLine($"\nAsserting block NextConsensus..{utBlock.NextConsensus}");
+            utBlock.NextConsensus.Should().Be(updatedContract.ScriptHash);
+
+            Console.WriteLine("\n==========================");
             // ============================================================================
             //                      finalize ConsensusService actor
             // ============================================================================
-
-            //Thread.Sleep(4000);
-            Sys.Stop(actorConsensus);
+            Console.WriteLine("Finalizing consensus service actor and returning states.");
             TimeProvider.ResetToDefault();
 
-            Assert.AreEqual(1, 1);
+            //Sys.Stop(actorConsensus);
+        }
+
+        public ConsensusPayload GetCommitPayloadModifiedAndSignedCopy(ConsensusPayload cpToCopy, ushort vI, KeyPair kp, byte[] blockHashToSign)
```

Comment:
Can we rename these parameters and make them more readable? Especially the first 2 params.

Reply:
In my opnion they have good names.
cpToCopy = consensusPayloadToCopy
vI = ValidatorIndex
kp = key pair
blockHashToSign = blockHashToSign

Do you suggest something else?

Reply:
Done as comments